dask_match/expr.py

```diff
@@ -1026,34 +962,30 @@ def _name(self):
     def _divisions(self):
         return self.exprs[0]._divisions()
 
-    def _fused_task(self):
-        block = {self._name: self.exprs[0]._name}
+    def _task(self, index):
+        graph = {self._name: (self.exprs[0]._name, index)}
         for _expr in self.exprs:
             if isinstance(_expr, Fused):
-                for k, v in _expr._fused_task().items():
-                    block[k] = v
+                (_, subgraph, name) = _expr._task(index)
+                graph.update(subgraph)
+                graph[(name, index)] = name
             else:
-                block[_expr._name] = _expr._blockwise_task()
-        return block
+                graph[(_expr._name, index)] = _expr._task(index)
 
-    def _layer(self):
-        # Create SubgraphCallable
-        dependencies = self.dependencies()
-        func = SubgraphCallable(
-            self._fused_task(),
-            self._name,
-            [dep._name for dep in dependencies],
+        for i, dep in enumerate(self.dependencies()):
+            graph[self._blockwise_arg(dep, index)] = "_" + str(i)
```

Comment:
Probably I should make these `_1`, `_2`, `_3` more unique, or try to reuse the existing keys (but without accidentally filling them)

Reply:
I'm okay with these names for now since they are easy to understand, and the only way for there to be a collision is if an `Expr` dependency used this name (which I would consider a bug in the implementation of that `Expr._name`).

Reply:
**Just to clarify**: `self.dependencies()` will only return `Expr` operands.